scripts/rock-paper-scissors-refactored.js

```diff
@@ -0,0 +1,82 @@
+// This seems like an easier way to keep track of what wins and what doesn't
+const choices = {
+  rock: {
+    beats: "scissors",
+  },
+  paper: {
+    beats: "rock",
+  },
+  scissors: {
+    beats: "paper",
+  },
+};
+
+// Returns an array of the keys of the object [rock, paper, scissors]
+const choiceCount = Object.keys(choices).length;
+
+function playRound(playerInput) {
+  const playerChoice = playerInput.toLowerCase();
+  
+  if (!choices[playerChoice]) {
+    addGameLog(`What is ${playerInput}? You can't do that! That's cheating!`);
+    return;
+  }
+
+  // Use Math.floor to round down to the nearest whole number.
+  // If we use Math.ceil, the array index could possibly be 3, which is out of bounds.
+  const computerChoice = Object.keys(choices)[Math.floor(Math.random() * choiceCount)];
+
+  if (playerChoice === computerChoice) {
+    addGameLog(`You Tied! ${playerChoice} is ${computerChoice}!`);
+    return;
+  }
+
+  const isPlayerWinner = choices[playerChoice].beats === computerChoice;
+  const resultPhrase = isPlayerWinner ? "You Win!" : "You Lose!";
+  const resultReason = isPlayerWinner ? "beats" : "loses to";
+  const resultString = `${resultPhrase} ${playerChoice} ${resultReason} ${computerChoice}!`;
+
+  updateScores(isPlayerWinner);
+  addGameLog(resultString);
+}
+
+function updateScores(isPlayerWinner) {
+  const playerScoreElement = document.getElementById("player-score");
+  const computerScoreElement = document.getElementById("computer-score");
+
+  // Convert the innerHTML to a number so that we can perform math ops on it
+  const playerScore = Number(playerScoreElement.innerHTML);
+  const computerScore = Number(computerScoreElement.innerHTML);
+
+  if (isPlayerWinner) {
+    playerScoreElement.innerHTML = playerScore + 1;
```

Comment:
Should we instead use innerText here? It seems like it should have the same effect while being marginally safer.

Reply:
`innerText` would also work, but I used `innerHTML` as a sort of way to "enforce" that the element will always _only_ contain a number value